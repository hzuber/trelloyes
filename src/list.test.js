import React from 'react';
import ReactDOM from 'react-dom';
import List from './list';
import Card from './card';
import renderer from 'react-test-renderer';

const testCards = [{ id: 'a', title: 'First card', content: 'lorem ipsum' },
{ id: 'b', title: 'Second card', content: 'lorem ipsum' },
{ id: 'c', title: 'Third card', content: 'lorem ipsum' },]

describe('list component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<List id='1' header='first list' cards= {testCards}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
        });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Card id='a' title='first card' content='lorem ipsum' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
