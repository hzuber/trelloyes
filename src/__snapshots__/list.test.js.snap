// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      first list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <button
    className="List-add-button"
    type="button"
  >
    + Add Random Card
  </button>
</section>
`;

exports[`list component renders the UI as expected 2`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    first card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
